src/kakarot/precompiles/blake2f.cairo

```diff
@@ -0,0 +1,573 @@
+// SPDX-License-Identifier: MIT
+
+%lang starknet
+
+// Starkware dependencies
+from starkware.cairo.common.alloc import alloc
+from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.cairo.common.cairo_builtins import BitwiseBuiltin
+from starkware.cairo.common.math import unsigned_div_rem
+from starkware.cairo.common.registers import get_fp_and_pc, get_label_location
+from starkware.cairo.common.math_cmp import is_nn, is_le
+
+// Internal dependencies
+from utils.utils import Helpers
+
+// @title Blake2f Precompile related functions.
+// @notice This file contains the logic required to run the blake2f precompile
+// @author @greged93
+// @custom:namespace PrecompileBlake2f
+namespace PrecompileBlake2f {
+    const PRECOMPILE_ADDRESS = 0x09;
+    const GFROUND = 1;
+
+    const ROUNDS_BYTES_LEN = 4;
+    const WORD_BYTES_LEN = 8;
+    const F_BYTES_LEN = 1;
+
+    const H_BYTES_OFFSET = 4;
+    const M_BYTES_OFFSET = 68;
+    const T_BYTES_OFFSET = 196;
+    const F_BYTES_OFFSET = 212;
+
+    // @notice Run the precompile.
+    // @param input_len The length of input array.
+    // @param input The input array.
+    // @return The output length, output array, and gas usage of precompile.
+    func run{
+        syscall_ptr: felt*,
+        pedersen_ptr: HashBuiltin*,
+        range_check_ptr,
+        bitwise_ptr: BitwiseBuiltin*,
+    }(input_len: felt, input: felt*) -> (output_len: felt, output: felt*, gas_used: felt) {
```

Comment:
I think all our precompiles should have the same signature.
I believe we added an "unused" `_address` field in order to enable error & event reporting

Reply:
Example: 

```cairo
namespace PrecompileEcRecover {
    const PRECOMPILE_ADDRESS = 0x01;
    const GAS_COST_EC_RECOVER = 3000;

    // @notice Run the precompile.
    // @param input_len The length of input array.
    // @param input The input array.
    // @return The output length, output array, and gas usage of precompile.
    func run{
        syscall_ptr: felt*,
        pedersen_ptr: HashBuiltin*,
        range_check_ptr,
        bitwise_ptr: BitwiseBuiltin*,
    }(_address: felt, input_len: felt, input: felt*) -> (
        output_len: felt, output: felt*, gas_used: felt
    ) {
```

Reply:
still need to fix this